apps/skilavottord/ws/src/app/modules/accessControl/accessControl.service.ts

```diff
@@ -1,69 +1,132 @@
+          },
+        ],
+      })
+      this.logger.info(
+        'findAll-recyclingUsers result:' + JSON.stringify(res, null, 2),
+      )
+      return res
+    } catch (error) {
+      this.logger.error('error finding all recyclingUsers:' + error)
+    }
+  }
 
-    // TODO replace mock data with actual db query
-    const res = await Promise.resolve([
-      {
-        nationalId: '1234567777',
-        name: 'Gervimaður',
-        role: Role.recyclingCompany,
-        partnerId: '8888888888',
-      },
-      {
-        nationalId: '1234567888',
-        name: 'Gervimaður2',
-        role: Role.recyclingCompany,
-        partnerId: '9999999999',
-      },
-      {
-        nationalId: '1234567899',
-        name: 'Gervimaður3',
-        role: Role.recyclingCompany,
-        partnerId: '9999999999',
-      },
-    ] as AccessControlModel[])
-    return res
+  async findOne(nationalId: string): Promise<AccessControlModel> {
+    this.logger.info('find one access user...')
+    try {
+      const res = await this.accessControlModel.findOne({
+        where: { nationalId: nationalId },
+        include: [
+          {
+            model: RecyclingPartnerModel,
+          },
+        ],
+      })
+      this.logger.info(
+        'findOne-accessControl result:' + JSON.stringify(res, null, 2),
+      )
+      // null if not found
+      return res
+    } catch (error) {
+      this.logger.error('error finding one AccessControl:' + error)
+    }
   }
 
+  // async createAccess(
+  //   input: CreateAccessControlInput,
+  // ): Promise<AccessControlModel> {
+  //   // TODO replace mock data with actual db query
+  //   return Promise.resolve({
+  //     nationalId: '1234567890',
+  //     name: 'Gervimaður3',
+  //     role: Role.recyclingCompany,
+  //     partnerId: '9999999999',
+  //   })
+  // }
+
   async createAccess(
     input: CreateAccessControlInput,
   ): Promise<AccessControlModel> {
-    // TODO replace mock data with actual db query
-    return Promise.resolve({
-      nationalId: '1234567890',
-      name: 'Gervimaður3',
-      role: Role.recyclingCompany,
-      partnerId: '9999999999',
-    })
+    this.logger.info('Creating Access:' + JSON.stringify(input, null, 2))
+    try {
+      const dd = input as AccessControlModel
+      const res = await dd.save()
+      if (res) {
+        return res as AccessControlModel
+      } else {
+        throw Error(`access could not be created.`)
+      }
+    } catch (error) {
+      this.logger.error('error creating Access:' + error)
+      throw Error('error creating Access:' + error)
+    }
   }
 
   async updateAccess(
     input: UpdateAccessControlInput,
   ): Promise<AccessControlModel> {
-    // TODO replace mock data with actual db query
-    return Promise.resolve({
-      nationalId: '1234567890',
-      name: 'Gervimaður3',
-      role: Role.recyclingCompany,
-      partnerId: '9999999999',
+    try {
+      const dd = await this.accessControlModel.findOne({
```

Comment:
Needs better naming.

Reply:
And this should call the existing AccessControlService method that already implements this feature...:
```typescript
const accessControl = await this.findOne(input.nationalId)
```